ngraph/core/reference/include/ngraph/runtime/reference/convolution_backprop_data.hpp

```diff
@@ -0,0 +1,309 @@
+                        size_t in_dilation_stride = in_dilation[i - 2];
+
+                        in_transform_start[i] = filter_movement_stride * out_coord[i];
+                        in_transform_end[i] = in_transform_start[i] +
+                                              (filter_shape[i] - 1) * filter_dilation_stride + 1;
+                        in_transform_movement_strides[i] = filter_dilation_stride;
+                        in_transform_pad_below[i] = below_pad;
+                        in_transform_pad_above[i] = above_pad;
+                        in_transform_dilation_strides[i] = in_dilation_stride;
+                    }
+
+                    AxisVector in_transform_axis_order(2 + n_spatial_dimensions);
+                    for (size_t i = 0; i < in_transform_axis_order.size(); i++)
+                    {
+                        in_transform_axis_order[i] = i;
+                    }
+                    CoordinateTransform in_transform(in_shape,
+                                                     in_transform_start,
+                                                     in_transform_end,
+                                                     in_transform_movement_strides,
+                                                     in_transform_axis_order,
+                                                     in_transform_pad_below,
+                                                     in_transform_pad_above,
+                                                     in_transform_dilation_strides);
+
+                    // Simultaneously with iterating I, for the filter we need to iterate the
+                    // coordinate:
+                    //
+                    //   F
+                    //
+                    // over the range (noninclusive on the right):
+                    //
+                    //   (chan_out,0,0,...,0) ->
+                    //     (chan_out+1,
+                    //      chans_in_count,
+                    //      filter_dims_1,
+                    //        ...,
+                    //      filter_dims_n)
+                    //
+                    // with unit stride.
+
+                    Shape filter_transform_start(2 + n_spatial_dimensions);
+                    Shape filter_transform_end(2 + n_spatial_dimensions);
+
+                    filter_transform_start[filter_out_channel_axis] = out_channel;
+                    filter_transform_end[filter_out_channel_axis] = out_channel + 1;
+                    filter_transform_start[filter_in_channel_axis] = 0;
+                    filter_transform_end[filter_in_channel_axis] = 1;
+
+                    for (size_t i = 2; i < n_spatial_dimensions + 2; i++)
+                    {
+                        filter_transform_start[i] = 0;
+                        filter_transform_end[i] = filter_shape[i];
+                    }
+
+                    CoordinateTransform filter_transform(
```

Comment:
ConvolutionBackpropData will be addressed in different ticket, so do not review it's code - I didn't change it, only moved it to other file because previously it was together with convolution.